wpilibc/src/main/native/include/frc/logging/LogFile.h

```diff
@@ -0,0 +1,63 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <ctime>
+#include <fstream>
+#include <string>
+
+namespace frc {
+
+/**
+ * A LogFile writes text to log in a file.
+ */
+class LogFile {
+ public:
+  /**
+   * Instantiate a LogFile passing in its prefix and its extension.
+   *
+   * If you want the file to be saved in a existing directory, you can add its
+   * path before the file prefix. Exemple : to save the file in a usb stick on
+   * the roborio ("/media/sda1/") : LogFile("/media/sda1/log").
+   *
+   * @param filePrefix The prefix of the LogFile.
+   * @param fileExtension The extension of the LogFile (without dot).
+   */
+  explicit LogFile(std::string filePrefix = "log",
```

Comment:
Use wpi::StringRef for string temporaries instead of std::string. It's in the `<wpi/StringRef.h>` header.